### PR TITLE
Fix mantis PR 7168 by creating a safety margin in the bytecode stack.

### PR DESCRIPTION
**Pull request by jhjourdan**
I use a safety margin of 60 words, which is enough for the event handler and the apply instructions.

### CODE REVIEW

**Comment by damiendoligez** [comment]
Minor nitpick: I'm not sure the constant should be defined in `config.mlp`, as I see no reason to believe that it will change with the underlying system.

**Comment by jhjourdan** [comment]
Indeed, it will not change with the underlying system. But nor does stack_threshold, and it may change if someone implements an instructions necessitating more space on the stack.

**Comment by damiendoligez** [comment]
@xavierleroy Could you review this patch?

File: testsuite/tests/misc/pr7168.ml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  let x = x+x in  let x = x+x in  let x = x+x in  let x = x+x in
+  let x = x+x in  let x = x+x in  let x = x+x in  let x = x+x in
+  let x = x+x in  let x = x+x in  let x = x+x in  let x = x+x in
+  let x = x+x in
+  let _ = f x in
+  ()
+
+let _ =
+  try f 1
+  with Stack_overflow -> Printf.printf "OK\n"
```
Comment (xavierleroy) [inline testsuite/tests/misc/pr7168.ml:71]: Can we please ensure that this test runs only in bytecode?  On some platforms, strange things can happen when ocamlopt-compiled code overflows the stack.

File: utils/config.mli [hunk 2]
```diff
@@ -90,6 +90,11 @@ val max_young_wosize: int
 val stack_threshold: int
         (* Size in words of safe area at bottom of VM stack,
            see byterun/config.h *)
+val stack_safety_margin: int
+        (* Size in words of the safety margin between the bottom of
+           the stack and the stack pointer. This margin can be used by
+           intermediate computations of some intructions, or the event
```
Comment (xavierleroy) [inline utils/config.mli:96]: Typo: "intructions"

**Comment by xavierleroy** [comment]
Looks good to me, provided the test is run in bytecode mode only.

**Comment by damiendoligez** [comment]
Fixed according to @xavierleroy 's remarks and merged into 4.03 (commit eda1b395dbcae9959b16a66924aa119223eeb0ca) and trunk (commit 1915e20c5b43cc7788c22027c6c76e265083cb97).

**Comment by jhjourdan** [comment]
Thanks, @damiendoligez !

**Comment by xavierleroy** [comment]
@jhjourdan: this PR is attracting new attention because of #11062.  I have one question: why 60 for the safety margin ? My analysis (below) suggests 6.

APPLY[123] +3
Setup_for_gc +3
Setup_for_c_call +2
Setup_for_event +6
Setup_for_debugger +4

**Comment by jhjourdan** [comment]
@xavierleroy : I don't have clear memories, but it really seems like it was based on nothing concrete. It just felt good to use a margin 10 times larger than what was empirically needed, since it was not clear to me which part of the interpreter could allocate out-of-the-limit stack slots.

**Pull request closed without merging**